server/notification-providers/gtx-messaging.js

```diff
@@ -0,0 +1,38 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+
+class GtxMessaging extends NotificationProvider {
+    name = "gtxmessaging";
+
+    /**
+     * @inheritDoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        const okMsg = "Sent Successfully.";
+
+        let authKey = notification.gtxMessagingApiKey;
+        let from = notification.gtxMessagingFrom.trim();
+        let to = notification.gtxMessagingTo.trim();
+        let text = msg.replaceAll("🔴 ", "").replaceAll("✅ ", "");
+
+        try {
+
+            let data = new URLSearchParams();
+            data.append("from", from);
+            data.append("to", to);
+            data.append("text", text);
+
+            let url = `https://rest.gtx-messaging.net/smsc/sendsms/${authKey}/json`;
+
+            console.log(`will post url: ${url}, data:`, data);
+
+            await axios.post(url, data);
```

Comment:
What is returned in the `message-status` field?

![image](https://github.com/louislam/uptime-kuma/assets/26258709/f639bd81-e9f7-44b2-81c1-221e2b3cc2c2)

Reply:
E.g.
```json
{
    "message-count": "1",
    "message-status": "OK",
    "message-id": "d593f2d6-de4a-48b2-8abe-17487d7f3c30e"
}
```

Reply:
I think a message should only be marked as successfully sent iff `response["message-status"] === "OK"`

Reply:
Again, I aligned my implementation with the other existing notification providers in uptime kuma. The ones I checked all discard the results from the API provider. E.g. 
- https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/pushover.js
- https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/rocket-chat.js
- https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/signal.js
- https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/slack.js
- https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/smsmanager.js

I appreciate your thoroughness but begin to wonder why things that seem to be totally ok in other providers are not ok in this one.
- https://github.com/louislam/uptime-kuma/pull/4323/files#diff-f6180734e4150ee5a5f4b32123d331b2d423a982bdc6dc1eac6270d9f15fc3eaR25
- https://github.com/louislam/uptime-kuma/pull/4323/files#diff-f6180734e4150ee5a5f4b32123d331b2d423a982bdc6dc1eac6270d9f15fc3eaR32
- https://github.com/louislam/uptime-kuma/pull/4323/files#diff-498441c7c64c3a7b0678c35933cf84efc6b44b78f1e4382bbfeaf3c91c21df53R22
- https://github.com/louislam/uptime-kuma/pull/4219/files#diff-349e3a3248bc86bc5ef940634eaba8d6b9291698ceb753fd68cc6ac975e9e956R70

I can add a check for the message status. However the API documentation doesn't state which values can be returned for `message-status`.  I could only return a message relaying the message-status. Without knowing the possible values we run the risk of reporting false-positives (e.g messages could be queued up resulting in status like `DELAYED` or `QUEUED`).

If you insist I will add the check. I personally see no need at this time since the API documentation doesn't describe error cases with an HTTP 200 response. I tested with invalid phone numbers. These also return HTTP 200 with message status `OK`.

Reply:
> but begin to wonder why things that seem to be totally ok in other providers are not ok in this one.

This depends on the thoroughness of the code review.
The review comments you in an OSS project might not always reflect the current state of the code (while useful for the initial draft of a PR), but rather the level of software-quality the project wants to be at.
Here this came up because I looked at the API docs and saw that 200 might not indicate success as normally thought.